core/src/block.rs

```diff
@@ -482,15 +482,19 @@ impl VersionedCandidateBlock {
     /// # Errors
     /// Forward errors from [`CandidateBlock::revalidate`]
     #[inline]
-    pub fn revalidate(
+    pub fn revalidate<const IS_GENESIS: bool>(
```

Comment:
was this my influence? :)

Reply:
yes :) 
I think that we know exactly places in code where we expect genesis transaction so it would be convenient to use const generics.

Reply:
I applaud this approach. Although I personally think that `::<false>` doesn't really convey the idea.

Reply:
I agree, and they should be on stable soon. So we might consider using them right now